app/controllers/posts_controller.rb

```diff
@@ -9,12 +9,18 @@ def new
   end
 
   def create
-    @post = current_user.posts.new(post_params)
-    if @post.save
-      redirect_to "/users/#{current_user.id}/posts", notice: 'Post created Successfully'
+    if current_user.nil?
+      flash[:alert] = 'You need to be logged in to create a post.'
+      redirect_to new_user_session_path
     else
-      flash[:alert] = 'Something went wrong'
-      render :new
+      @post = current_user.posts.new(post_params)
+
+      if @post.save
+        redirect_to user_posts_path(current_user), notice: 'Post created successfully.'
+      else
+        flash.now[:alert] = 'Something went wrong.'
+        render :new
+      end
     end
   end
 
```

Comment:
- [ ] [REQUIRED] You have done a great job so far 🥇 however, a part of the project requirement states that you need to use CanCanCan for the Post authorization so that a user can delete a post and a comment if it is theirs or if they have an admin role. This is supposed to be implemented in the PostsController.
Take a look at this article for [hint](https://www.rubydoc.info/github/CanCanCommunity/cancancan/CanCan%2FControllerAdditions%2FClassMethods:load_and_authorize_resource)